mantidimaging/gui/windows/operations/presenter.py

```diff
@@ -275,12 +275,24 @@ def _post_filter(self, updated_stacks: list[ImageStack], task):
                     # otherwise check with user which one to keep
                     if self.view.safeApply.isChecked():
                         use_new_data = self._wait_for_stack_choice(stack, stack.id)
-                    # if the stack that was kept happened to have a proj180 stack - then apply the filter to that too
-                    if stack.has_proj180deg() and use_new_data and not self.applying_to_all:
+
+                    if self.is_a_proj180deg(stack):
+                        # Handle 180-degree stacks based on the log_180_degree attribute
+                        if not self.model.selected_filter.log_180_degree:
+                            if task.progress:
+                                task.progress("Skipping normalization for the 180-degree stack.")
+                            continue
+                    # Apply to proj180 synchronously if necessary
+                    if use_new_data and not self.applying_to_all and not self.is_a_proj180deg(stack):
                         # Apply to proj180 synchronously - this function is already running async
                         # and running another async instance causes a race condition in the parallel module
                         # where the shared data can be removed in the middle of the operation of another operation
-                        self._do_apply_filter_sync([stack.proj180deg])
+                        try:
+                            self._do_apply_filter_sync([stack.proj180deg])
+                        except RuntimeError as e:
+                            self.view.show_error_dialog(f"Operation failed on proj180deg stack: {e}")
+                            continue
```

Comment:
I don't think we need the extra try/except. The message is being propagated fine currently.